plugin/docker/container.go

```diff
@@ -68,29 +66,30 @@ func (c *container) Exec(ctx context.Context, cmd string, args []string, opts pl
 	}
 	created, err := c.client.ContainerExecCreate(ctx, c.id, cfg)
 	if err != nil {
-		return execResult, err
+		return nil, err
 	}
 
 	resp, err := c.client.ContainerExecAttach(ctx, created.ID, types.ExecStartCheck{})
 	if err != nil {
-		return execResult, err
+		return nil, err
 	}
 
-	// Asynchronously copy container exec output to an exec output channel.
-	outputCh, stdout, stderr := plugin.CreateExecOutputStreams(ctx)
-	go func() {
-		_, err := stdcopy.StdCopy(stdout, stderr, resp.Reader)
-		activity.Record(ctx, "Exec on %v complete: %v", c.Name(), err)
-		stdout.CloseWithError(err)
-		stderr.CloseWithError(err)
+	cmdObj := plugin.NewRunningCommand(ctx)
+	cmdObj.SetStopFunc(func() {
```

Comment:
I figured moving this closer to `cmdObj` would make the "This command is currently running, here's how we stop it" part more obvious. Can move it back if it just makes things more confusing.